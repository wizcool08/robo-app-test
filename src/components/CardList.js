import React from "react";
import Card from "./Card";

const CardList = ({ robots }) => {
  console.log("CardList");
  return (
    <div className="list-items">
      {robots.map((robot, i) => {
        return (
          <Card className='robot' key={i} id={robot.id} name={robot.name} email={robot.email} />
        );
      })}
    </div>
  );
};

export default CardList;
