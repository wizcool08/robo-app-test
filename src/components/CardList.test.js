import { shallow } from "enzyme";
import React from "react";
import CardList from "./CardList";

it("render the CardList Component", () => {
  const mockRobots = [
    {
      id: 1,
      name: "John Snow",
      username: "Johnny Walker",
      email: "john@gmail.com"
    },
    {
      id: 2,
      name: "Snow White",
      username: "SkyWalker",
      email: "snow@gmail.com"
    }
  ];

  const wrapper = shallow(<CardList robots={mockRobots} />);
  expect(wrapper.debug()).toMatchSnapshot();
});
