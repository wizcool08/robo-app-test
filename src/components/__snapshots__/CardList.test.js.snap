// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the CardList Component 1`] = `
"<div className=\\"list-items\\">
  <Card className=\\"robot\\" id={1} name=\\"John Snow\\" email=\\"john@gmail.com\\" />
  <Card className=\\"robot\\" id={2} name=\\"Snow White\\" email=\\"snow@gmail.com\\" />
</div>"
`;
